### PR TITLE
Style design: Nav bar with Infinite Scroll and Group Convo by Date

### PR DESCRIPTION
**Pull request by walbercardoso**
## Summary
Change the conversations list in the SideBar, grouping Convos by: Today, Last 7 days, Last 30 days, and year

Before:

image

After:
image

## Change Type

Please delete any irrelevant options.

- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [x] My code adheres to this project's style guidelines
- [x] I have performed a self-review of my own code
- [x] I have commented in any complex areas of my code
- [x] I have made pertinent documentation changes
- [x] My changes do not introduce new warnings
- [x] I have written tests demonstrating that my changes are effective or that my feature works
- [x] Local unit tests pass with my changes
- [x] Any changes dependent on mine have been merged and published in downstream modules.

### CODE REVIEW

File: client/src/components/Conversations/Conversations.tsx [hunk 1]
```diff
@@ -2,6 +2,54 @@ import Convo from './Convo';
...
+  if (isWithinInterval(date, { start: subDays(now, 30), end: now })) {
+    return 'Last 30 days';
+  }
+  return ' ' + getYear(date).toString(); // Returns the year for anything older than 30 days
+};
+
+// Function to group conversations
+const groupConversationsByDate = (conversations) => {
+  const groups = conversations.reduce((acc, conversation) => {
+    const date = parseISO(conversation.createdAt);
```
Comment (danny-avila) [inline client/src/components/Conversations/Conversations.tsx]: Use the `updatedAt` field, this is more in line with the official version

Comment (walbercardoso) [reply client/src/components/Conversations/Conversations.tsx]: Right.

**Comment by danny-avila** [comment]
Are you at a spot where I can take a look at the new convo/deleted convo issue?

**Comment by walbercardoso** [comment]
Its in the infinite scroll implementation. You want me to include in this same PR or make a New for it?

**Comment by walbercardoso** [comment]
Updated. Including infinite loop. New convos is now being displayed in the convos list, just the title is not updating

**Comment by walbercardoso** [comment]
Closed by mistake. Updated with all cleaned code. Sorry

**Comment by walbercardoso** [comment]
Wondering if would be better to hide scroll bar

**Comment by fuegovic** [comment]
> Wondering if would be better to hide scroll bar

I tested briefly and it looks like you fixed the issue with the new convo titles 🎉

About the scroll bar, I think it's good to have it but the position could be optimized a little bit, maybe make it more like on the official OpenAI website (the current lack of space between the title-box and the scroll bar makes it look out of place imo)

**LibreChat**
image

**OpenAI**
image

Overall great job, this is going to be a nice improvement over the current system! 🔥

**Comment by walbercardoso** [comment]
> > Wondering if would be better to hide scroll bar
> 
> I tested briefly and it looks like you fixed the issue with the new convo titles 🎉
> 
> About the scroll bar, I think it's good to have it but the position could be optimized a little bit, maybe make it more like on the official OpenAI website (the current lack of space between the title-box and the scroll bar makes it look out of place imo)
> 
> **LibreChat** image
> 
> **OpenAI** image
> 
> Overall great job, this is going to be a nice improvement over the current system! 🔥

Thank you @fuegovic. I will improve the layout with your suggestion

**Comment by fuegovic** [comment]
@walbercardoso I tested the latest changes you made, it's looking good! 💫

**Comment by walbercardoso** [comment]
Also, it improved a fade effect of the convos 

image

**Comment by Berry-13** [comment]
> I think its done:
> 
> image
> 
> Icon changes based on Endpoint selected

why is the icon so big?

**Comment by danny-avila** [comment]
Amazing work @walbercardoso 

As @Berry-13 mentioned, there could be better formatting for the icon in the New Chat button

On official when hovering:
image

**Comment by walbercardoso** [comment]
> Amazing work @walbercardoso
> 
> As @Berry-13 mentioned, there could be better formatting for the icon in the New Chat button
> 
> On official when hovering: image

Agree... gonna improve that

**Comment by walbercardoso** [comment]
image

Its done

**Comment by danny-avila** [comment]
@walbercardoso thanks so much for this contribution. I'm going to sprint this one out as I'd like to address the following items myself.

There's a couple of errors in the browser logs:

> Warning: Cannot update a component (`Nav`) while rendering a different component (`Landing`). To locate the bad setState() call inside `Landing`, follow the stack trace as described in https://reactjs.org/link/setstate-in-render

> Uncaught (in promise) Error: Unknown endpoint: Mistral

This second is due to my configuration but some extra steps you didn't know about need to handle it

**Next, in the network logs. The fetching should not be so frequent** 

**LibreChat**
image


**ChatGPT:**
image

page 1 is being invalidated every time. also these requests should be throttled as they can easily be abused right now. these are also fetching again on refocus, which is a legacy thing but I'll removing it with this update

**Lastly some styling items:**

Would like to more closely mirror still (LibreChat on left)

<div style="text-align: center;">
    
    
</div>


And the spinner should be part of the convo items like in ChatGPT:

image

**Comment by danny-avila** [comment]
Closing for #1708

**Comment by walbercardoso** [comment]
@danny-avila It's always a pleasure to contribute. I started programming in react because of Librechat and since then it has been very rewarding to learn a little more from you all. Love this project.

**Comment by danny-avila** [comment]
> @danny-avila It's always a pleasure to contribute. I started programming in react because of Librechat and since then it has been very rewarding to learn a little more from you all. Love this project.

Thanks man! Greatly appreciate you taking the time to contribute. I just mean to take over to include these changes in the assistants branch on my schedule. It also gives me the impetus to implement this long awaited styling much sooner than I would've without you.

**Pull request closed without merging**